sklearn/inspection/_partial_dependence.py

```diff
@@ -81,30 +86,59 @@ def _grid_from_X(X, percentiles, grid_resolution):
 
     values = []
     for feature in range(X.shape[1]):
-        uniques = np.unique(_safe_indexing(X, feature, axis=1))
-        if uniques.shape[0] < grid_resolution:
-            # feature has low resolution use unique vals
-            axis = uniques
-        else:
-            # create axis based on percentiles and grid resolution
-            emp_percentiles = mquantiles(
-                _safe_indexing(X, feature, axis=1), prob=percentiles, axis=0
-            )
-            if np.allclose(emp_percentiles[0], emp_percentiles[1]):
+        if feature in custom_values:
```

Comment:
It would make sense to rename `feature` to `feature_idx` since it's not the feature name.

Reply:
`custom_values` is also index-based, and perhaps the name should reflect that (since `custom_values` is also an argument in the `partial_dependence` function, and has a different structure.